src/protocols/secure_channel/PairingSession.cpp

```diff
@@ -138,6 +138,17 @@ CHIP_ERROR PairingSession::DecodeMRPParametersIfPresent(TLV::Tag expectedTag, TL
     return tlvReader.ExitContainer(containerType);
 }
 
+bool PairingSession::IsSessionEstablishmentInProgress()
+{
+    if (!mSecureSessionHolder)
+    {
+        return false;
+    }
+
+    Transport::SecureSession * secureSession = mSecureSessionHolder->AsSecureSession();
+    return secureSession->IsPairing();
```

Comment:
What does "is pairing" mean?

Reply:
From https://github.com/project-chip/connectedhomeip/blob/master/src/transport/SecureSession.h#L233:
```
        // kPending denotes a secure session object that is internally
        // reserved by the stack before and during session establishment.
        //
        // Although the stack can tolerate eviction of these (releasing one
        // out from under the holder would exhibit as CHIP_ERROR_INCORRECT_STATE
        // during CASE or PASE), intent is that we should not and would leave
        // these untouched until CASE or PASE complete.
        //
        // During this stage, the reference counter is hold by the PairingSession
```

In an earlier version I was trying to keep track of if we need to perform the abort within CASESession itself, but I found that we can just rely on the state inside `SecureSession`. I can go back to what I had before I found this if you think it is better to not rely on the possible state of this thing that is a few levels away